### PR TITLE
Add UODE Lotka-Volterra notebook

### PR DESCRIPTION
**Pull request by markusheinonen**
- Observe both prey and predator (H=I_2) at dt=0.2 (150 points)
- Laplace(0, 0.005) prior for SINDy-like sparsity on Theta
- SVI config: lr=1e-3, cov_rescaling=2.0, lr_decay=0.5, 2000 steps
- Correctly recovers xy interaction coefficients (Frobenius err ~0.077)
- Includes debug CLI script for systematic experiments

### CODE REVIEW

**Comment by DanWaxman** [comment]
Just got this to be working again and fixed some text/formatting issues. I think it makes sense to add to our deep dives for now, so I deleted the debug script and added this to our docs site nav. LGTM. Since I just made a bunch of changes, I think it's best if Matt does a quick review of this before merging. Thanks @markusheinonen!!

**Comment by mattlevine22** [comment]
@copilot Do a thorough check of the Jupyter notebook and please provide a bulleted list of how to fix the exposition to match the code (there are many discrepancies). For example, we code `H=jnp.eye(state_dim)` but write about partial observations and `H = [1  0]` in the exposition (also learning rate and cov_rescaling are not used in the code anymore).

**Review by mattlevine22 — APPROVED** [review]
Looks good, thanks! Fixed the text so it matches what was done (no LR schedule, no cov_rescaling, full observations, EnKF not EKF).

Tried to get copilot to do it, but that was a big mistake lol.

**Comment by DanWaxman** [comment]
Haha thanks Matt!! I tried but I guess I missed some of them… and thanks
again for drafting it, Markus!
